src/map.rs

```diff
@@ -1091,6 +1091,30 @@ where
         None
     }
 
+    /// Retains only the elements specified by the predicate.
+    ///
+    /// In other words, remove all pairs (k, v) such that f(&k,&v) returns false.
+    pub fn retain<F>(&mut self, mut f: F)
+        where
+            F: FnMut(&K, &V) -> bool
```

Comment:
[`std::HashMap::retain`](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.retain) uses `FnMut(&K, &mut V) -> bool` for filtering the map. But in our case we do not have a `get_mut` or similar method, so I changed the signature. I'm not sure if this is ok.

Reply:
No, `(&K, &V)` seems fine since we are working with a concurrent data structure after all!